main.tf

```diff
@@ -191,11 +216,15 @@ resource "azurerm_kubernetes_cluster" "main" {
       condition     = (var.client_id != "" && var.client_secret != "") || (var.identity_type == "SystemAssigned") || (var.identity_ids == null ? false : length(var.identity_ids) > 0)
       error_message = "If use identity and `UserAssigned` or `SystemAssigned, UserAssigned` is set, an `identity_ids` must be set as well."
     }
+    precondition {
+      condition     = (var.microsoft_defender_enabled == true && var.log_analytics_workspace_enabled == true) || var.microsoft_defender_enabled == false
```

Comment:
Would the following code be better?:

```hcl
condition     = !var.microsoft_defender_enabled || var.log_analytics_workspace_enabled
```

Reply:
Or you can use the following expression:

```hcl
condition     = !(var.microsoft_defender_enabled && !var.log_analytics_workspace_enabled)
```

That'll be easier to understand, right?